client/src/controllers/DrilldownController.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Instantiates an a11y dialog on the controlled element.
+ * Adds support for hide and show methods and blocking body
+ * scroll when the dialog is open.
+ *
+ * @example
+ * <div
+ *    data-controller="w-dialog"
+ *    data-w-dialog-theme-value="floating"
+ *   >
+ *    <div data-w-dialog-target="body"></div>
+ *  </div>
+ */
+export class DrilldownController extends Controller<HTMLElement> {
+  static targets = ['menu', 'toggle', 'back'];
+
+  declare readonly menuTarget: HTMLElement;
+  declare readonly toggleTargets: HTMLButtonElement[];
+  declare readonly backTargets: HTMLButtonElement[];
+
+  connect() {
+    this.toggleTargets.forEach((toggle) => {
+      toggle.addEventListener('click', this.open.bind(this));
+    });
+
+    this.backTargets.forEach((back) => {
+      back.addEventListener('click', this.close.bind(this));
+    });
+  }
+
+  open(e: MouseEvent) {
+    const toggle = e.target as HTMLButtonElement;
+    this.toggle(true, toggle);
+  }
+
+  close() {
+    this.toggleTargets.forEach((toggle) => this.toggle(false, toggle));
+    this.element.focus();
+  }
+
+  toggle(expanded: boolean, toggle: HTMLButtonElement) {
+    const content = this.element.querySelector<HTMLElement>(
+      `#${toggle.getAttribute('aria-controls')}`,
+    );
+    if (content) {
+      toggle.setAttribute('aria-expanded', expanded.toString());
+      content.hidden = !expanded;
+      this.menuTarget.hidden = expanded;
+      this.element.classList.toggle('w-drilldown--active', expanded);
```

Comment:
Might be nice to have this class come in via the Stimulus classes.

```ts

static classes = ['active'];

//...
      this.element.classList.toggle(this.activeClass, expanded);

// OR if you want to be defensive (not throw an error if no activeClass set.
      if (this.hasActiveClass) {
        this.element.classList.toggle(this.activeClass, expanded);
      }
```

Then in the HTML

```html
                                <div class="w-drilldown" tabindex="-1" data-controller="w-drilldown" data-w-drilldown-active-class=""w-drilldown--active">
```

This also lets the classes/styling be isolated from the JS code and means the classes are all in the same place (HTML template).

Reply:
This class isn’t used so it will just be removed